From ee3be974167b62f91c2e4b9f64f05fa91ebcba15 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Wed, 17 Nov 2021 12:45:44 +0100
Subject: [PATCH 2/2] setup-sshd: use SSH_KEY instead of prompt user

Do not ask user for ssh key, but support the use of SSH_KEY env var.
---
 setup-sshd.in | 20 +++++++-------------
 1 file changed, 7 insertions(+), 13 deletions(-)

diff --git a/setup-sshd.in b/setup-sshd.in
index caf8f1c..271ad79 100644
--- a/setup-sshd.in
+++ b/setup-sshd.in
@@ -18,6 +18,7 @@ usage() {
 	exit 1
 }
 
+authorized_key="$SSH_KEY"
 while getopts "hc:k:" opt; do
 	case $opt in
 		h) usage;;
@@ -53,21 +54,14 @@ if [ -n "$svc" ]; then
 	rc-service $svc start
 fi
 
-if [ -z "$authorized_key" ]; then
-	ask "Authorized SSH public key for root? (HTTP(S)/FTP URL or the public key itself)" none
-	authorized_key="$resp"
-fi
-
 if [ -n "$authorized_key" -a "$authorized_key" != "none" ]; then
 	# if the argument is an HTTP(S)/FTP URL, try to fetch the file contents
-	if [ -z "$(echo "$authorized_key" | sed -E 's~^(https?|ftp)://.+$~~')" ]; then
-		key_url="$authorized_key"
-		authorized_key="$(wget -qO- "$key_url")" || die "Could not fetch key from '$key_url'"
-
-		echo "Received authorized SSH key from '$key_url':"
-		echo "$authorized_key"
-	fi
-
+	case "$authorized_key" in
+		http*://*|ftp://)
+			key_url="$authorized_key"
+			authorized_key="$(wget -qO- "$key_url")" || die "Could not fetch key from '$key_url'"
+			;;
+	esac
 	mkdir -p ${ROOT}/root/.ssh
 	echo "$authorized_key" >> ${ROOT}/root/.ssh/authorized_keys
 fi
-- 
2.34.0

